### PR TITLE
TF-838 Refactor `RefreshChange` to sync `State`

### PR DESCRIPTION
**Pull request by dab246**
#838 

### Resolved

https://user-images.githubusercontent.com/80730648/184799279-6c63cc14-ddda-4405-b4d5-4879b0787b87.mov

### CODE REVIEW

File: lib/features/thread/data/repository/thread_repository_impl.dart [hunk 1]
```diff
@@ -314,10 +314,6 @@ class ThreadRepositoryImpl extends ThreadRepository {
           filter: EmailFilterCondition(inMailbox: trashMailboxId, before: lastEmail?.receivedAt),
           properties: Properties({EmailProperty.id}));
 
-      if (emailsResponse.state != null) {
-        await _updateState(emailsResponse.state!);
-      }
-
```
Comment (hoangdat) [inline lib/features/thread/data/repository/thread_repository_impl.dart:320]: 👍

**Comment by hoangdat** [comment]
I think this PR deserves with 1 ADR :) Thank for hardworking ❤️

**Review by ManhNTX — APPROVED** [review]

**Review by ManhNTX — APPROVED** [review]
LGTM

File: docs/adr/0014-refactor-refresh-change-to-sync-state.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+Accepted
+
+## Context
+
+- `State` is updated asynchronously on the server so after performing operations on `Email` the data is not fully updated.
+
+## Decision
+
+- Pass the initial `State` to the result after performing operations with `Email`
+- Use `State` in the returned results to perform `RefreshChange` of `Email` and `Mailbox`
```
Comment (hoangdat) [inline docs/adr/0014-refactor-refresh-change-to-sync-state.md:16]: ❤️

**Review by hoangdat — APPROVED** [review]

**Pull request merged**